cast/src/lib.rs

```diff
@@ -31,10 +31,22 @@ pub fn cast<S: AsRef<str>>(
         let cast_target = map.iter().find(|x| x.0.as_ref() == field.name());
 
         let col = match cast_target {
-            Some(tgt) => Arc::new(
-                arrow::compute::cast_with_options(col, &tgt.1, &cast_opt)
-                    .with_context(|| format!("Failed when casting column '{}'", field.name()))?,
-            ),
+            Some(tgt) => {
+                // allow precision loss for decimal types into floating point types
+                if matches!(col.data_type(), DataType::Decimal256(..) | DataType::Decimal128(..)) && tgt.1.is_floating() {
+                    let string_col = arrow::compute::cast_with_options(col, &DataType::Utf8, &cast_opt)
+                        .with_context(|| format!("Failed when casting column '{}' to string as intermediate step", field.name()))?;
+                    Arc::new(
+                        arrow::compute::cast_with_options(&string_col, &tgt.1, &cast_opt)
+                            .with_context(|| format!("Failed when casting column '{}' to {:?}", field.name(), tgt.1))?,
+                    )
+                } else {
+                    Arc::new(
+                        arrow::compute::cast_with_options(col, &tgt.1, &cast_opt)
+                            .with_context(|| format!("Failed when casting column '{}' from {:?} to {:?}", field.name(), col.data_type(), tgt.1))?,
+                    )
+                }
+            }
```

Comment:
Arrow doesn't allow cast Decimals to Float straight. Using this hack to cast as string first, so that cast is made even with precision lost.

We can add another param to cast `alow_precision_loss`, but I don't think it's necessary, this is expected behavior.